pyvista/core/filters/image_data.py

```diff
@@ -1022,6 +1039,27 @@ def points_to_cells(self, scalars: str | None = None, *, copy: bool = True):
             By default, all point data arrays at the input are passed through as cell
             data at the output.
 
+        dimensionality : bool, VectorLike[bool], Literal['0D', '1D', '2D', '3D', 'preserve'], default: 'preserve'
+            Control which dimensions will be modified by the filter.
+
+            - ``True``: all dimensions are modified.
+            - ``False``: no dimensions are modified.
```

Comment:
> I'm not sure if these options should be included... isn't `True` equivalent to `'3D'`?

It's correct that `True` is equivalent to `'3D'`. However, I consider `'1D'`, `'2D'`, `'3D'`, etc. as aliases for boolean vectors.

> [At] least one dimension _must_ change for 1D inputs, at least two for 2D, and all three for 3D, the conversion to be valid?

Indeed, there is no filter for which `False` makes sense except `cells_to_points` and `points_to_cells` for 0D input, but I considered it for the sack of generability (a future filter could require it?). For the 2D and 3D inputs, it is, for instance, possible to pad only one dimension and get a valid output.

Reply:
> > I'm not sure if these options should be included... isn't `True` equivalent to `'3D'`?
> 
> If we have boolean vectors, I think it is nice to automatically broadcast `True` and `False` but it might not be required to document it.

I like your general implementation to use boolean vectors. My main concern is with the broadcasting specifically and allowing `True` and `False` scalars. This is because I find `dimensionality=True` unclear from a plain-language perspective... unless you look at the docs it's not immediately clear what this means. `dimensionality='3D'` is much clearer I think, so why not just use that instead? And, forcing a vector in this case (e.g. `dimensionality=(True,True,True)` hints that a mask is being specified, otherwise using simply `True` or `False` could be (incorrectly) interpreted as an option that is being enabled.

> > [At] least one dimension _must_ change for 1D inputs, at least two for 2D, and all three for 3D, the conversion to be valid?
> 
> Indeed, there is no filter for which `False` makes sense except `cells_to_points` and `points_to_cells` for 0D input, but I considered it for the sack of generability (a future filter could require it?). 

This would still be supported by specifying `dimensionality=(False, False, False)` explicitly. Again, I like the generalization, but not sure if we should be supporting the broadcasting.

> For the 2D and 3D inputs, it is, for instance, possible to pad only one dimension and get a valid output.

In these cases, isn't it necessary to provide the full length-3 mask? e.g. `(True, True, False)`? These cases are still supported if the broadcasting option is removed.

Reply:
That makes sense, thanks for your input.